Build failing on Mac OS 10.12.6

Build fails with the following when running `ninja`
![image](https://user-images.githubusercontent.com/5173131/33524792-2e39eb3e-d81b-11e7-9deb-f2d012cfd4e4.png)

![image](https://user-images.githubusercontent.com/5173131/33524797-4ac8f4d4-d81b-11e7-9756-eeeeba100d9c.png)

Output from `meson . build`:
![image](https://user-images.githubusercontent.com/5173131/33524804-68460100-d81b-11e7-9570-cf7c5ab01a1d.png)

Happy to provide more info.
I'm currently trying to build kiwix-lib locally, which depends on libzim.

I now get the following:
```
(libzim) ➜  build git:(master) ✗ ninja
[2/40] Compiling C++ object 'src/zim@sha/file.cpp.o'.
FAILED: src/zim@sha/file.cpp.o
c++  -Isrc/zim@sha -Isrc -I../src -Iinclude -I../include -I/usr/local/Cellar/xz/5.2.3/include -I/usr/local/Cellar/xapian/1.4.5/include -I/usr/local/Cellar/icu4c/59.1_1/include -Xclang -fcolor-diagnostics -pipe -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++11 -O0 -g -Werror -Wall -pthread -MMD -MQ 'src/zim@sha/file.cpp.o' -MF 'src/zim@sha/file.cpp.o.d' -o 'src/zim@sha/file.cpp.o' -c ../src/file.cpp
In file included from ../src/file.cpp:23:
../include/zim/search.h:35:12: error: class 'InternalData' was previously declared as a struct [-Werror,-Wmismatched-tags]
    friend class search_iterator::InternalData;
           ^
../include/zim/search_iterator.h:60:16: note: previous use is here
        struct InternalData;
               ^
1 error generated.
[7/40] Compiling C++ object 'src/zim@sha/file
```

--- a/include/zim/search.h
+++ b/include/zim/search.h
@@ -32,7 +32,7 @@ class File;
 class Search
 {
     friend class search_iterator;
-    friend class search_iterator::InternalData;
+    friend struct search_iterator::InternalData;
     public:
         typedef search_iterator iterator;
 

